### PR TITLE
MINOR; Various code cleanups

### PR DESCRIPTION
**Pull request by dajac**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/unit/kafka/network/ConnectionQuotasTest.scala [hunk 1]
```diff
@@ -782,14 +783,14 @@ class ConnectionQuotasTest {
   }
 
   private def verifyNoBlockedPercentRecordedOnAllListeners(): Unit = {
-    blockedPercentMeters.foreach { case (name, meter) =>
+    blockedPercentMeters.forKeyValue { (name, meter) =>
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/network/ConnectionQuotasTest.scala:786]: line#809 and line#761 have similar issue. Could you fix them in this PR?

**Review by chia7712 — APPROVED** [review]
LGTM

There are many `foreach` in ``ReassignPartitionsCommand`. How about replacing all of them in PR?

**Comment by dajac** [comment]
@chia7712 That makes sense. I have updated the PR.

File: core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala [hunk 2]
```diff
@@ -515,15 +512,14 @@ object ReassignPartitionsCommand extends Logging {
...
-        val replicasForTopic = zkClient.getReplicaAssignmentForTopics(Set(topic))
-        partitions.foreach {
-          case (partition, targetReplicas) =>
-            val currentReplicas = replicasForTopic.getOrElse(partition, Seq())
-            results.put(partition, new PartitionReassignmentState(
-              currentReplicas, targetReplicas, !partitionsBeingReassigned.contains(partition)))
-        }
+    targetReassignments.groupBy(_._1.topic).forKeyValue { (topic, partitions) =>
+      val replicasForTopic = zkClient.getReplicaAssignmentForTopics(Set(topic))
+      partitions.foreach {
```
Comment (chia7712) [inline core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala]: `forKeyValue`

**Review by chia7712 — APPROVED** [review]
+1 after fixing a trivial issue :)

**Comment by dajac** [comment]
@chia7712 I just pushed a commit to address your comment. That commit also improves a catch clause.

**Review by chia7712 — APPROVED** [review]
+1

**Pull request merged**